### PR TITLE
emacs 27.2

### PR DESCRIPTION
**Pull request by timvisher**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/emacs.rb [hunk 1]
```diff
@@ -4,24 +4,9 @@ class Emacs < Formula
   license "GPL-3.0-or-later"
 
   stable do
-    url "https://ftp.gnu.org/gnu/emacs/emacs-27.1.tar.xz"
-    mirror "https://ftpmirror.gnu.org/emacs/emacs-27.1.tar.xz"
-    sha256 "4a4c128f915fc937d61edfc273c98106711b540c9be3cd5d2e2b9b5b2f172e41"
-
-    # The emacs binary is patched with a signature after linking. This invalidates the code
```
Comment (timvisher) [inline Formula/emacs.rb:11]: I noticed the warning that these patches may have been resolved upstream. I removed them and ran the install successfully but it also looks like these patches may be for situations I may not necessarily have encountered. I'm building on 10.14.6 on Intel. I'm happy to put these back if it looks necessary.

Comment (carlocab) [reply Formula/emacs.rb:11]: Did you try building with the patches? If they fail to apply, then that's a sign we don't need them anymore.

If both patches can be removed, then we don't need the `stable do` block anymore either; `url`, `mirror`, and `sha256` can be placed back near the top of the formula.

Comment (timvisher) [reply Formula/emacs.rb:11]: I'll try it with the patches in place. :)

Comment (timvisher) [reply Formula/emacs.rb:11]: Seems like the fail?

```diff
diff --git a/Formula/emacs.rb b/Formula/emacs.rb
index 44632bf083..a721360cd4 100644
--- a/Formula/emacs.rb
+++ b/Formula/emacs.rb
@@ -7,6 +7,21 @@ class Emacs < Formula
     url "https://ftp.gnu.org/gnu/emacs/emacs-27.2.tar.xz"
     mirror "https://ftpmirror.gnu.org/emacs/emacs-27.2.tar.xz"
     sha256 "b4a7cc4e78e63f378624e0919215b910af5bb2a0afc819fad298272e9f40c1b9"
+
+    # The emacs binary is patched with a signature after linking. This invalidates the code
+    # signature. Code signing is required on Apple Silicon. This patch adds a step to resign
+    # the binary after it is patched.
+    patch do
+      url "https://github.com/emacs-mirror/emacs/commit/868f51324ac96bc3af49a826e1db443548c9d6cc.patch?full_index=1"
+      sha256 "d2b19fcca66338d082c15fa11d57abf7ad6b40129478bef4c6234c19966db988"
+    end
+
+    # Back-ported patch for configure and configure.guess to allow configure to complete
+    # for aarch64-apple-darwin targets.
+    patch do
+      url "https://raw.githubusercontent.com/Homebrew/formula-patches/25c1e1797d4004a9e5b9453779399afc63d04b97/emacs/arm.patch"
+      sha256 "5f812fc413b722e294c7f7abd38f3a9bbda84ec68537cea42900a81e57c7ecb1"
+    end
   end
 
   bottle do

```


```
$ brew reinstall --build-from-source emacs
Warning: Treating emacs as a formula. For the cask, use homebrew/cask/emacs
==> Downloading https://github.com/emacs-mirror/emacs/commit/868f51324ac96bc3af49a826e1db443548c9d6cc.patch?full_index=1
######################################################################## 100.0%
==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/25c1e1797d4004a9e5b9453779399afc63d04b97/emacs/arm.pa
######################################################################## 100.0%
==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-27.2.tar.xz
Already downloaded: /Users/timothyvisher/Library/Caches/Homebrew/downloads/2cd8fd48748257712575a817f0cb133837fd4fbd6f5d6f7c402401af9f862e32--emacs-27.2.tar.xz
==> Reinstalling emacs
Warning: A newer Command Line Tools release is available.
…

==> Patching
==> Applying 868f51324ac96bc3af49a826e1db443548c9d6cc.patch
patching file src/Makefile.in
Hunk #1 succeeded at 341 with fuzz 2 (offset 5 lines).
Hunk #2 FAILED at 662.
1 out of 2 hunks FAILED -- saving rejects to file src/Makefile.in.rej

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
emacs 27.2 https://github.com/Homebrew/homebrew-core/pull/73912
```

Comment (carlocab) [reply Formula/emacs.rb:11]: Try installing twice, each time with only one of the patches in place, so we're sure that both patches don't apply.

Comment (timvisher) [reply Formula/emacs.rb:11]: Looks like neither one applies anymore.

    $ git -C $(brew --repository homebrew/core) diff
    diff --git a/Formula/emacs.rb b/Formula/emacs.rb
    index 44632bf083..f6111cc344 100644
    --- a/Formula/emacs.rb
    +++ b/Formula/emacs.rb
    @@ -7,6 +7,21 @@ class Emacs < Formula
         url "https://ftp.gnu.org/gnu/emacs/emacs-27.2.tar.xz"
         mirror "https://ftpmirror.gnu.org/emacs/emacs-27.2.tar.xz"
         sha256 "b4a7cc4e78e63f378624e0919215b910af5bb2a0afc819fad298272e9f40c1b9"
    +
    +    # # The emacs binary is patched with a signature after linking. This invalidates the code
    +    # # signature. Code signing is required on Apple Silicon. This patch adds a step to resign
    +    # # the binary after it is patched.
    +    # patch do
    +    #   url "https://github.com/emacs-mirror/emacs/commit/868f51324ac96bc3af49a826e1db443548c9d6cc.patch?full_index=1"
    +    #   sha256 "d2b19fcca66338d082c15fa11d57abf7ad6b40129478bef4c6234c19966db988"
    +    # end
    +
    +    # Back-ported patch for configure and configure.guess to allow configure to complete
    +    # for aarch64-apple-darwin targets.
    +    patch do
    +      url "https://raw.githubusercontent.com/Homebrew/formula-patches/25c1e1797d4004a9e5b9453779399afc63d04b97/emacs/arm.patch"
    +      sha256 "5f812fc413b722e294c7f7abd38f3a9bbda84ec68537cea42900a81e57c7ecb1"
    +    end
       end
    
       bottle do
    
    $ brew reinstall --build-from-source emacs
    Warning: Treating emacs as a formula. For the cask, use homebrew/cask/emacs
    ==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/25c1e1797d4004a9e5b9453779399afc63d04b97/emacs/arm.patch
    Already downloaded: /Users/timothyvisher/Library/Caches/Homebrew/downloads/9de031be9d941bc27749d8f7f91e8e0e1a7c588e9b73894d820e14871fe12909--arm.patch
    ==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-27.2.tar.xz
    Already downloaded: /Users/timothyvisher/Library/Caches/Homebrew/downloads/2cd8fd48748257712575a817f0cb133837fd4fbd6f5d6f7c402401af9f862e32--emacs-27.2.tar.xz
    ==> Reinstalling emacs
    Warning: A newer Command Line Tools release is available.
    …
    
    ==> Patching
    ==> Applying arm.patch
    patching file build-aux/config.guess
    Hunk #1 succeeded at 2 with fuzz 2.
    patching file configure
    Hunk #1 FAILED at 4978.
    1 out of 1 hunk FAILED -- saving rejects to file configure.rej
    patching file configure.ac
    Hunk #1 FAILED at 703.
    1 out of 1 hunk FAILED -- saving rejects to file configure.ac.rej
    
    READ THIS: https://docs.brew.sh/Troubleshooting
    
    These open issues may also help:
    emacs 27.2 https://github.com/Homebrew/homebrew-core/pull/73912
    
    Error: A newer Command Line Tools release is available.
    …
    
    
    $ git -C $(brew --repository homebrew/core) diff
    diff --git a/Formula/emacs.rb b/Formula/emacs.rb
    index 44632bf083..3e0a5996d1 100644
    --- a/Formula/emacs.rb
    +++ b/Formula/emacs.rb
    @@ -7,6 +7,21 @@ class Emacs < Formula
         url "https://ftp.gnu.org/gnu/emacs/emacs-27.2.tar.xz"
         mirror "https://ftpmirror.gnu.org/emacs/emacs-27.2.tar.xz"
         sha256 "b4a7cc4e78e63f378624e0919215b910af5bb2a0afc819fad298272e9f40c1b9"
    +
    +    # The emacs binary is patched with a signature after linking. This invalidates the code
    +    # signature. Code signing is required on Apple Silicon. This patch adds a step to resign
    +    # the binary after it is patched.
    +    patch do
    +      url "https://github.com/emacs-mirror/emacs/commit/868f51324ac96bc3af49a826e1db443548c9d6cc.patch?full_index=1"
    +      sha256 "d2b19fcca66338d082c15fa11d57abf7ad6b40129478bef4c6234c19966db988"
    +    end
    +
    +    # # Back-ported patch for configure and configure.guess to allow configure to complete
    +    # # for aarch64-apple-darwin targets.
    +    # patch do
    +    #   url "https://raw.githubusercontent.com/Homebrew/formula-patches/25c1e1797d4004a9e5b9453779399afc63d04b97/emacs/arm.patch"
    +    #   sha256 "5f812fc413b722e294c7f7abd38f3a9bbda84ec68537cea42900a81e57c7ecb1"
    +    # end
       end
    
       bottle do
    
    $ brew reinstall --build-from-source emacs
    Warning: Treating emacs as a formula. For the cask, use homebrew/cask/emacs
    ==> Downloading https://github.com/emacs-mirror/emacs/commit/868f51324ac96bc3af49a826e1db443548c9d6cc.patch?full_index=1
    Already downloaded: /Users/timothyvisher/Library/Caches/Homebrew/downloads/378b3b9d72aed62ef412623056105bce90fc3eecb7c765716fb11c79942752ba--868f51324ac96bc3af49a826e1db443548c9d6cc.patch
    ==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-27.2.tar.xz
    Already downloaded: /Users/timothyvisher/Library/Caches/Homebrew/downloads/2cd8fd48748257712575a817f0cb133837fd4fbd6f5d6f7c402401af9f862e32--emacs-27.2.tar.xz
    ==> Reinstalling emacs
    Warning: A newer Command Line Tools release is available.
    …
    
    ==> Patching
    ==> Applying 868f51324ac96bc3af49a826e1db443548c9d6cc.patch
    patching file src/Makefile.in
    Hunk #1 succeeded at 341 with fuzz 2 (offset 5 lines).
    Hunk #2 FAILED at 662.
    1 out of 2 hunks FAILED -- saving rejects to file src/Makefile.in.rej
    
    READ THIS: https://docs.brew.sh/Troubleshooting
    
    These open issues may also help:
    emacs 27.2 https://github.com/Homebrew/homebrew-core/pull/73912
    
    Error: A newer Command Line Tools release is available.
    …

Comment (carlocab) [reply Formula/emacs.rb:11]: Great, thanks for looking into this!

**Review by carlocab — APPROVED** [review]
Thanks for your first contribution to homebrew/core, @timvisher.

**Pull request closed without merging**